pkg/oc/cli/cmd/image/mirror/mirror.go

```diff
@@ -98,47 +102,49 @@ type pushOptions struct {
 	MaxPerRegistry int
 
 	AttemptS3BucketCopy []string
+
+	genericclioptions.IOStreams
+	resource.FilenameOptions
 }
 
-// schema2ManifestOnly specifically requests a manifest list first
-var schema2ManifestOnly = distribution.WithManifestMediaTypes([]string{
-	manifestlist.MediaTypeManifestList,
-	schema2.MediaTypeManifest,
-})
+func NewMirrorImageOptions(streams genericclioptions.IOStreams) *MirrorImageOptions {
+	return &MirrorImageOptions{
+		IOStreams:      streams,
+		MaxRegistry:    4,
+		MaxPerRegistry: 6,
+	}
+}
 
 // NewCommandMirrorImage copies images from one location to another.
-func NewCmdMirrorImage(name string, out, errOut io.Writer) *cobra.Command {
-	o := &pushOptions{}
-
+func NewCmdMirrorImage(name string, streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewMirrorImageOptions(streams)
 	cmd := &cobra.Command{
 		Use:     "mirror SRC DST [DST ...]",
 		Short:   "Mirror images from one repository to another",
 		Long:    mirrorDesc,
 		Example: fmt.Sprintf(mirrorExample, name+" mirror"),
 		Run: func(c *cobra.Command, args []string) {
-			o.Out = out
-			o.ErrOut = errOut
 			kcmdutil.CheckErr(o.Complete(args))
 			kcmdutil.CheckErr(o.Run())
 		},
 	}
 
-	flag := cmd.Flags()
-	flag.BoolVar(&o.DryRun, "dry-run", o.DryRun, "Print the actions that would be taken and exit without writing to the destinations.")
-	flag.BoolVar(&o.Insecure, "insecure", o.Insecure, "Allow push and pull operations to registries to be made over HTTP")
-	flag.BoolVar(&o.SkipMount, "skip-mount", o.SkipMount, "Always push layers instead of cross-mounting them")
-	flag.BoolVar(&o.SkipMultipleScopes, "skip-multiple-scopes", o.SkipMultipleScopes, "Some registries do not support multiple scopes passed to the registry login.")
-	flag.StringVar(&o.FilterByOS, "filter-by-os", o.FilterByOS, "A regular expression to control which images are mirrored. Images will be passed as '<platform>/<architecture>[/<variant>]'.")
-	flag.BoolVar(&o.Force, "force", o.Force, "Attempt to write all layers and manifests even if they exist in the remote repository.")
-	flag.IntVar(&o.MaxRegistry, "max-registry", 4, "Number of concurrent registries to connect to at any one time.")
-	flag.IntVar(&o.MaxPerRegistry, "max-per-registry", 6, "Number of concurrent requests allowed per registry.")
-	flag.StringSliceVar(&o.AttemptS3BucketCopy, "s3-source-bucket", o.AttemptS3BucketCopy, "A list of bucket/path locations on S3 that may contain already uploaded blobs. Add [store] to the end to use the Docker registry path convention.")
-	flag.StringSliceVarP(&o.Filenames, "filename", "f", o.Filenames, "One or more files to read SRC=DST or SRC DST [DST ...] mappings from.")
```

Comment:
This change looks suspicious. You sure the -f for this command means the same thing?

Reply:
Not sure what you mean, the `f` shorthand is now added by the call to `kcmdutil.AddJsonFilenameFlag`